internal/metrics/metrics.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+)
+
+type Metrics struct {
+	PolicyEvaluationDurationMilliseconds *prometheus.HistogramVec
+}
+
+func SetupMetrics(prefix string) Metrics {
+	m := Metrics{
+		PolicyEvaluationDurationMilliseconds: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: prefix,
+			Name:      "policy_evaluation_duration_milliseconds",
+			Help:      "A histogram of the policy evaluation durations in milliseconds.",
+			Buckets:   []float64{1, 5, 10, 50, 100},
```

Comment:
since most policies may run in the order of microseconds wouldn't the 1ms bucket be too big?

Reply:
I thought about the metrics I would be interested to know if my policy has an impact on my http call.
I supposed that for an api call, it would be great a policy evaluation of less than 1ms (but maybe we could set also 0.5ms), and for evaluation of more than 100ms it would be a performance issue (but also in this case, maybe we could add other data).

What do you think?

Reply:
I agree, let's keep it this way.